Compilation on g++ < 8 broken due to filesystem header

```
/workspace/OpenMS/src/openms/source/SYSTEM/File.cpp:52:10: fatal error: filesystem: No such file or directory
 #include <filesystem>
```

Introduced in #7035 

https://cdash.openms.de/viewBuildError.php?buildid=148485

Please update min compiler version or ifdef the include to "experimental" in g++7

--- a/src/openms/source/SYSTEM/File.cpp
+++ b/src/openms/source/SYSTEM/File.cpp
@@ -50,9 +50,6 @@
 
 #include <atomic>
 
-// https://github.com/OpenMS/OpenMS/issues/7051
-//#include <filesystem>
-
 #ifdef OPENMS_WINDOWSPLATFORM
 #include <Windows.h> // for GetCurrentProcessId() && GetModuleFileName()
 #endif
@@ -176,14 +173,11 @@ namespace OpenMS
     return fi.exists() && fi.isExecutable();
   }
 
-  UInt64 File::fileSize(const String& /* file */)
+  UInt64 File::fileSize(const String& file)
   {
-    std::error_code err;
-    // https://github.com/OpenMS/OpenMS/issues/7051
-    //UInt64 res = std::filesystem::file_size(file.c_str(), err);
-    UInt64 res = 0;
-    if (err) return -1;
-    return res;
+    if (!File::exists(file)) return -1;
+
+    return QFile(file.toQString()).size();
   }
 
   bool File::rename(const String& from, const String& to, bool overwrite_existing, bool verbose)

--- a/src/tests/class_tests/openms/source/File_test.cpp
+++ b/src/tests/class_tests/openms/source/File_test.cpp
@@ -72,6 +72,12 @@ START_SECTION((static bool empty(const String &file)))
   TEST_EQUAL(File::empty(OPENMS_GET_TEST_DATA_PATH("File_test_text.txt")), false)
 END_SECTION
 
+START_SECTION((static UInt64 fileSize(const String& file)))
+  TEST_EQUAL(File::fileSize("does_not_exists.txt"), -1)
+  TEST_EQUAL(File::fileSize(OPENMS_GET_TEST_DATA_PATH("File_test_empty.txt")), 0)
+  TEST_EQUAL(File::fileSize(OPENMS_GET_TEST_DATA_PATH("File_test_text.txt")), 15)
+END_SECTION
+
 START_SECTION((static bool remove(const String &file)))
   //deleting non-existing file
   TEST_EQUAL(File::remove("does_not_exists.txt"), true)

